tesseract_motion_planners/ompl/src/ompl_motion_planner.cpp

```diff
@@ -348,33 +240,10 @@ PlannerResponse OMPLMotionPlanner::solve(const PlannerRequest& request) const
     }
 
     // Extract Solution
-    const std::vector<std::string> joint_names = manip->getJointNames();
-    const Eigen::MatrixX2d joint_limits = manip->getLimits().joint_limits;
-    tesseract_common::TrajArray traj = toTrajArray(simple_setup->getSolutionPath(), extractor);
-    assert(checkStartState(simple_setup->getProblemDefinition(), traj.row(0), extractor));
-    assert(checkGoalState(simple_setup->getProblemDefinition(), traj.bottomRows(1).transpose(), extractor));
-    assert(traj.rows() >= num_output_states);
-
-    // Enforce limits
-    for (Eigen::Index i = 0; i < traj.rows(); i++)
-    {
-      assert(tesseract_common::satisfiesLimits<double>(traj.row(i), joint_limits, 1e-4));
-      tesseract_common::enforceLimits<double>(traj.row(i), joint_limits);
-    }
+    CompositeInstruction output = cur_plan_profile->convertPath(simple_setup->getSolutionPath(), composite_mi, request.env);
 
-    // Assign trajectory to results
-    start_index = assignTrajectory(response.results,
-                                   start_move_instruction.getUUID(),
-                                   end_move_instruction.getUUID(),
-                                   start_index,
-                                   joint_names,
-                                   traj,
-                                   request.format_result_as_input);
-
-    // Reset data for next segment
-    start_instruction = end_instruction;
-    num_output_states = 1;
-    ++segment;
+    // Concatenate into results
+    std::copy(response.results.end(), output.begin(), output.end());
```

Comment:
This is where I need a little help understanding what exactly `assignTrajectory` was doing so we can replicate that behavior correctly

Reply:
The intent to is to preserve as much of the structure between two constrained waypoints. There can be any number of interpolation waypoints that should be preserved.